src/framework/vst/internal/synth/vstsynthesiser.cpp

```diff
@@ -0,0 +1,163 @@
+#include "vstsynthesiser.h"
+
+#include "log.h"
+
+#include "internal/vstplugin.h"
+
+using namespace mu;
+using namespace mu::vst;
+
+VstSynthesiser::VstSynthesiser(const VstPluginMeta& meta)
+    : m_pluginMeta(meta), m_vstAudioClient(std::make_unique<VstAudioClient>())
+{
+}
+
+Ret VstSynthesiser::init()
+{
+    RetVal plugin = repository()->findPluginById(m_pluginMeta.id);
+
+    if (!plugin.ret) {
+        return plugin.ret;
+    }
+
+    m_pluginPtr = plugin.val;
+
+    m_vstAudioClient->init(m_pluginPtr->component());
+
+    return make_ret(Ret::Code::Ok);
+}
+
+bool VstSynthesiser::isValid() const
+{
+    if (!m_pluginPtr) {
+        return false;
+    }
+
+    return m_pluginPtr->isValid();
+}
+
+bool VstSynthesiser::isActive() const
+{
+    return m_isActive;
+}
+
+void VstSynthesiser::setIsActive(bool arg)
+{
+    m_isActive = arg;
+}
+
+std::string VstSynthesiser::name() const
+{
+    return m_pluginMeta.name;
+}
+
+audio::synth::SoundFontFormats VstSynthesiser::soundFontFormats() const
+{
+    NOT_SUPPORTED;
+    return { audio::synth::SoundFontFormat::Undefined };
+}
+
+Ret VstSynthesiser::addSoundFonts(const std::vector<io::path>& /*sfonts*/)
+{
+    NOT_SUPPORTED;
+    return Ret(Ret::Code::NotSupported);
+}
+
+Ret VstSynthesiser::removeSoundFonts()
+{
+    NOT_SUPPORTED;
+    return Ret(Ret::Code::NotSupported);
+}
+
+bool VstSynthesiser::handleEvent(const midi::Event& e)
+{
+    if (!m_vstAudioClient) {
+        return false;
+    }
+
+    return m_vstAudioClient->handleEvent(e);
+}
+
+void VstSynthesiser::writeBuf(float* stream, unsigned int samples)
+{
+    m_vstAudioClient->process(stream, samples);
+}
+
+void VstSynthesiser::allSoundsOff()
+{
+    NOT_IMPLEMENTED;
+}
+
+void VstSynthesiser::flushSound()
+{
+    NOT_IMPLEMENTED;
+}
+
+Ret VstSynthesiser::setupChannels(const std::vector<midi::Event>& /*events*/)
+{
+    NOT_IMPLEMENTED;
+    return Ret(Ret::Code::Ok);

```

Comment:
I think need just send events to m_vstAudioClient (call m_vstAudioClient->handleEvent(e))

Reply:
yeah, I'm going to care about "channels" in the next PRs, maybe we can also add functionality for channel management (volume, balance, etc.). I just need to dive deeper first